plugins/vli/fu-vli-usbhub-firmware.c

```diff
@@ -50,7 +50,7 @@ fu_vli_usbhub_firmware_parse(FuFirmware *firmware,
 {
 	FuVliUsbhubFirmware *self = FU_VLI_USBHUB_FIRMWARE(firmware);
 	gsize bufsz = 0;
-	guint16 adr_ofs = 0;
+	guint32 adr_ofs = 0;
```

Comment:
you sure that's safe? isn't adr_ofs being used in other places as a 16 bit number? It might be safer to create a new variable, e.g. adr_ofs32.

Reply:
Quick note, he got pulled into a meeting just now but we'll make the necessary changes in due time. I believe he was testing and forgot to change the name.

Reply:
It's ok using a 32 bit variable to save 16 bit value or 24 bit value.

Reply:
I agree, but it's not okay to increase the alignment of a pointer on architectures other than x86_64.

Reply:
Previously in file `fu-vli-usbhub-firmware.c`, line 117,  `fu_memread_uint16_safe()`, we need to read 2 bytes from buffer to adr_ofs; now in file `fu-vli-usbhub-firmware.c`, line 100,  we need to read 3 bytes from buffer to adr_ofs. We thought `fu_memread_uint24_safe()` would work for this purpose. If not, can you help us find a better approach?

Reply:
GCC even warns you here:

	../plugins/vli/fu-vli-usbhub-firmware.c: In function ‘fu_vli_usbhub_firmware_parse’:
	../plugins/vli/fu-vli-usbhub-firmware.c:117:65: error: passing argument 4 of ‘fu_memread_uint16_safe’ from incompatible pointer type [-Werror=incompatible-pointer-types]
	  117 |                 if (!fu_memread_uint16_safe(buf, bufsz, 0x8000, &adr_ofs, G_BIG_ENDIAN, error)) {
	      |                                                                 ^~~~~~~~
	      |                                                                 |
	      |                                                                 guint32 * {aka unsigned int *}
	In file included from ../libfwupdplugin/fwupdplugin.h:79,
			 from ../plugins/vli/fu-vli-usbhub-firmware.h:10,
			 from ../plugins/vli/fu-vli-usbhub-firmware.c:11:
	../libfwupdplugin/fu-mem.h:47:33: note: expected ‘guint16 *’ {aka ‘short unsigned int *’} but argument is of type ‘guint32 *’ {aka ‘unsigned int *’}
	   47 |                        guint16 *value,
	      |                        ~~~~~~~~~^~~~~
	../plugins/vli/fu-vli-usbhub-firmware.c:209:65: error: passing argument 4 of ‘fu_memread_uint16_safe’ from incompatible pointer type [-Werror=incompatible-pointer-types]
	  209 |                 if (!fu_memread_uint16_safe(buf, bufsz, 0x8018, &adr_ofs, G_BIG_ENDIAN, error)) {
	      |                                                                 ^~~~~~~~
	      |                                                                 |
	      |                                                                 guint32 * {aka unsigned int *}
	../libfwupdplugin/fu-mem.h:47:33: note: expected ‘guint16 *’ {aka ‘short unsigned int *’} but argument is of type ‘guint32 *’ {aka ‘unsigned int *’}
	   47 |                        guint16 *value,
	      |                        ~~~~~~~~~^~~~~

I'll push a trivial fix for you.

Reply:
Sorry I wrote it wrong. Thanks for your help.